server/expression/types/types.go

```diff
@@ -0,0 +1,32 @@
+package types
+
+import "regexp"
+
+type Type uint
+
+const (
+	TYPE_NIL       Type = 0
```

Comment:
In go, it's usually preferred to have constant following the [same format as variables](https://stackoverflow.com/a/22688926). Also, you can use `iota` here for easier maintenance (https://golangbyexample.com/iota-in-golang/).

I would rewrite this block as:

```
const (
	TypeNil Type = iota
	TypeString
	TypeNumber
	TypeAttribute
	TypeDuration
)
```